arangod/Agency/MoveShard.cpp

```diff
@@ -298,6 +299,19 @@ bool MoveShard::start(bool&) {
     return false;
   }
 
+  // Compute group to move shards together:
+  std::vector<Job::shard_t> shardsLikeMe =
+      clones(_snapshot, _database, _collection, _shard);
+
+  if (found < 0) {   // not in Plan, then it must not be a failoverCandidate:
+    auto failoverCands = Job::findAllFailoverCandidates(
+        _snapshot, _database, shardsLikeMe);
+    if (failoverCands.find(_to) != failoverCands.end()) {
+      finish("", "", false, "toServer must not be in failoverCandidates for shard or any of its distributeShardsLike colleagues");
```

Comment:
Should this finish the job or just wait and do nothing?

Reply:
That is a good question. I do not like stuck jobs. Therefore I opted for abortion. But I can understand the argument that the client might be happier if we wait until the server is out of `failoverCandidates` and then do his bidding.

Reply:
In all likelyhood, this will not make a difference in real life...